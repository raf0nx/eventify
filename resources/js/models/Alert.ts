import { EnumAlertType } from "@/enums/EnumAlertType";

export class AlertModel {
    showAlert = false;
    type = EnumAlertType.SUCCESS;
    message = "";
    dismissible = false;
    callback: Function | null = null;
    btnText: string = "";

    setShowAlert(showAlert: boolean): AlertModel {
        this.showAlert = showAlert;
        return this;
    }

    setType(type: EnumAlertType): AlertModel {
        this.type = type;
        return this;
    }

    setMessage(message: string): AlertModel {
        this.message = message;
        return this;
    }

    setDismissible(dismissible: boolean): AlertModel {
        this.dismissible = dismissible;
        return this;
    }

    setCallback(callback: Function): AlertModel {
        this.callback = callback;
        return this;
    }

    setBtnText(text: string): AlertModel {
        this.btnText = text;
        return this;
    }
}
