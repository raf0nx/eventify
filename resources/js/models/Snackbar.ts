import { EnumSnackbarColor } from "@/enums/EnumSnackbarColor";
import { EnumSnackbarIcon } from "@/enums/EnumSnackbarIcon";

export class SnackbarModel {
    showSnackbar = false;
    message = "";
    icon = EnumSnackbarIcon.SUCCESS;
    color = EnumSnackbarColor.SUCCESS;

    setShowSnackbar(value: boolean): SnackbarModel {
        this.showSnackbar = value;
        return this;
    }

    setMessage(message: string): SnackbarModel {
        this.message = message;
        return this;
    }

    setIcon(icon: EnumSnackbarIcon): SnackbarModel {
        this.icon = icon;
        return this;
    }

    setColor(color: EnumSnackbarColor): SnackbarModel {
        this.color = color;
        return this;
    }
}
