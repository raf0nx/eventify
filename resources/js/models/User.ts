export interface User {
    id: Number;
    name: String;
    email: string;
    created_at: Date;
    email_verified_at?: Date;
    updated_at: Date;
}
