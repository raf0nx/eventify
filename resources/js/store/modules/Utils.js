export const namespaced = true;

export const state = {
    showSnackbar: false,
    showAlert: false,
    loading: false
};
export const mutations = {
    SET_ALERT(state, setAlert) {
        state.showAlert = setAlert;
    },
    SET_LOADING(state, isLoading) {
        state.loading = isLoading;
    },
    SET_SNACKBAR(state, setSnackbar) {
        state.showSnackbar = setSnackbar;
    }
};
export const actions = {
    setSnackbar({ commit }) {
        commit("SET_SNACKBAR", true);
        setTimeout(() => commit("SET_SNACKBAR", false), 2500);
    }
};

export const getters = {
    showAlert: state => {
        return state.showAlert;
    },
    isLoading: state => {
        return state.loading;
    },
    showSnackbar: state => {
        return state.showSnackbar;
    }
};
