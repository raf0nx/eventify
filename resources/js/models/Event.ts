import { User } from "./User";

export interface Event {
    created_at: Date;
    description: string;
    event_image: string;
    id: number;
    name: string;
    start_datetime: Date;
    updated_at: Date;
    users: User[];
}
