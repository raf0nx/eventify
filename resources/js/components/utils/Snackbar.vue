<template>
	<v-snackbar
		v-model="snackbar.showSnackbar"
        class="mb-8"
		app
		fixed
		bottom
		:color="snackbar.color"
	>
		<div class="d-flex align-center">
			<v-icon color="white">{{ snackbar.icon }}</v-icon>
			<span class="font-weight-bold ml-2">{{ snackbar.message }}</span>
		</div>
	</v-snackbar>
</template>

<script lang="ts">
	import { Vue, Component } from "vue-property-decorator";

	import { UtilsModule } from "@modules/Utils";
	import { SnackbarModel } from "@/models/Snackbar";

	@Component
	export default class Snackbar extends Vue {
		get snackbar(): SnackbarModel {
			return UtilsModule.snackbar;
		}
	}
</script>

<style scoped>
</style>