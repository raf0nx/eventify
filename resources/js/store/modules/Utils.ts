import {
    Module,
    VuexModule,
    Mutation,
    Action,
    getModule
} from "vuex-module-decorators";

import store from "@/store/store";
import { SnackbarModel } from "@/models/Snackbar";
import { AlertModel } from "@/models/Alert";

export interface UtilsState {
    snackbarNotification: SnackbarModel;
    alertNotification: AlertModel;
    isLoading: boolean;
    navDrawer: boolean;
}

@Module({ dynamic: true, store, name: "utils" })
class Utils extends VuexModule implements UtilsState {
    snackbarNotification = new SnackbarModel();
    alertNotification = new AlertModel();
    isLoading = false;
    navDrawer = false;

    @Mutation
    private SET_ALERT(alert: AlertModel): void {
        this.alertNotification = alert;
    }

    @Mutation
    private SET_LOADER(isLoading: boolean): void {
        this.isLoading = isLoading;
    }

    @Mutation
    private SET_SNACKBAR(snackbar: SnackbarModel): void {
        this.snackbarNotification = snackbar;
    }

    @Mutation
    private SET_NAV_DRAWER(setNavDrawer: boolean): void {
        this.navDrawer = setNavDrawer;
    }

    @Action
    setAlert(alert: AlertModel): void {
        this.SET_ALERT(alert);
    }

    @Action
    setLoader(isLoading: boolean): void {
        this.SET_LOADER(isLoading);
    }

    @Action
    setSnackbar(snackbar: SnackbarModel): void {
        this.SET_SNACKBAR(snackbar);
    }

    @Action
    setNavDrawer(setNavDrawer: boolean): void {
        this.SET_NAV_DRAWER(setNavDrawer);
    }

    get alert(): AlertModel {
        return this.alertNotification;
    }

    get loader(): boolean {
        return this.isLoading;
    }

    get snackbar(): SnackbarModel {
        return this.snackbarNotification;
    }

    get isNavDrawer(): boolean {
        return this.navDrawer;
    }
}

export const UtilsModule = getModule(Utils);
