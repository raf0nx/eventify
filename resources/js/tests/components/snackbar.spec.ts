import Vuetify from "vuetify";
import { createLocalVue, shallowMount, Wrapper } from "@vue/test-utils";

import Snackbar from "@components/utils/Snackbar.vue";
import { UtilsModule } from "@modules/Utils";
import { SnackbarModel } from "@/models/Snackbar";

describe("Snackbar Component", () => {
    const localVue = createLocalVue();

    let vuetify: Vuetify;
    let wrapper: Wrapper<Snackbar>;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = shallowMount(Snackbar, {
            localVue,
            vuetify
        });

        UtilsModule.setSnackbar(new SnackbarModel());
    });

    afterEach(() => {
        wrapper.destroy();
    });

    it("Should get snackbar object from store", () => {
        // Arrange
        const snackbarObj = UtilsModule.snackbar;

        // Assert
        // @ts-ignore
        expect(wrapper.vm.snackbar).toEqual(snackbarObj);
    });
});
