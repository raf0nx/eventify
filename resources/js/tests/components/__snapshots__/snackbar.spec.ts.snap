// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar Component Should match snapshot 1`] = `
"<v-snackbar-stub color=\\"deep-purple\\" tag=\\"div\\" bottom=\\"true\\" app=\\"true\\" contentclass=\\"\\" timeout=\\"5000\\" transition=\\"v-snack-transition\\" fixed=\\"\\" class=\\"mb-8\\" value=\\"true\\">
  <div class=\\"d-flex align-center\\">
    <v-icon-stub color=\\"white\\">mdi-check-circle</v-icon-stub> <span class=\\"font-weight-bold ml-2\\">Message for test purposes</span>
  </div>
</v-snackbar-stub>"
`;
